apps/ios/Shared/Views/UserSettings/PrivacySettings.swift

```diff
@@ -96,45 +199,126 @@ struct SimplexLockSetting: View {
             case .laFailedAlert: return laFailedAlert()
             case .laUnavailableInstructionAlert: return laUnavailableInstructionAlert()
             case .laUnavailableTurningOffAlert: return laUnavailableTurningOffAlert()
+            case .laPasswordSetAlert: return passwordAlert("Passcode set!")
+            case .laPasswordChangedAlert: return passwordAlert("Passcode changed!")
+            case .laPasswordNotChangedAlert: return mkAlert(title: "Passcode not changed!")
             }
         }
+        .sheet(item: $showPasswordAction) { a in
+            switch a {
+            case .enableAuth:
+                SetAppPasscodeView {
+                    laLockDelay = 30
+                    prefPerformLA = true
+                    showChangePassword = true
+                    showLAAlert(.laPasswordSetAlert)
+                } cancel: {
+                    prefPerformLA = false
+                    withAnimation { performLA = false }
+                    performLAToggleReset = true
+                }
+            case .toggleMode:
```

Comment:
not critical, but strangely asymmetric behavior between modes:
when switching from System auth to Passcode, if passcode entry is failed (cancelled), mode is reverted to System; when switching from Passcode to System, if system auth is failed lock is disabled

Reply:
maybe it's because password is removed on line 265? if so probably better so than to keep it longer